### Hypothesis

What will be the atipose tissue rate if waist size=28 ?

In [53]:
import pandas as pd

df = pd.read_csv(r"E:\DebugShala\Machine learning\6 ML intro_superviesd_unsup_&_Regression\wc-at.csv")
df

Waist      AT
0     74.75   25.72
1     72.60   25.89
2     81.80   42.60
3     83.95   42.80
4     74.65   29.84
..      ...     ...
104  100.10  124.00
105   93.30   62.20
106  101.80  133.00
107  107.90  208.00
108  108.50  208.00

[109 rows x 2 columns]

### Pre-Processing

In [54]:
df.isna().sum()

Waist    0
AT       0
dtype: int64

In [55]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 109 entries, 0 to 108
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Waist   109 non-null    float64
 1   AT      109 non-null    float64
dtypes: float64(2)
memory usage: 1.8 KB


1) there is no null value so no need to fill
2) ther is no string or text so need need to convert into numeric .. LabelEncoder not needed
3) values are already small so no need to scale down using MinMaxScaler

In [56]:
# splitting into train test

from sklearn.model_selection import train_test_split

train,test = train_test_split(df,train_size=0.8)

In [57]:
df.shape

(109, 2)

In [58]:
train.shape

(87, 2)

In [59]:
test.shape

(22, 2)

In [60]:
train_x = train.iloc[:,:1] #independent variable ---> feature
train_y = train.iloc[:,1] #dependent variable ---> target

In [61]:
train_x.shape

(87, 1)

In [62]:
train_y.shape

(87,)

In [63]:
test_x = test.iloc[:,:1] #independent variable
test_y = test.iloc[:,1] #dependent variable

In [64]:
test_x.shape

(22, 1)

In [65]:
test_y.shape

(22,)

### Simple Linear regression

In [66]:
from sklearn.linear_model import LinearRegression

In [67]:
slr = LinearRegression().fit(train_x,train_y)
prd = slr.predict(test_x)

In [68]:
prd

array([110.81661646, 158.30448206, 151.57112798, 200.83092887,
        19.9163364 ,  40.82517275,  41.71114039, 110.81661646,
       148.02725742, 101.95694004,  80.69371663,  30.5479481 ,
        37.45849571,  50.92520387,  44.9006239 ,  35.33217337,
       165.74661025, 133.85177515, 126.76403401,  61.20242851,
       160.4308044 , 110.81661646])

In [69]:
test_y

93      80.95
107    208.00
96     241.00
65     106.00
15      55.78
0       25.72
11      43.86
35     121.00
95     125.00
25      70.40
28      78.89
5       21.68
18      43.35
60      57.05
19      29.31
9       32.22
81     126.00
103    188.00
97     134.00
27      84.30
87     159.00
47     123.00
Name: AT, dtype: float64

In the predicted output we have not much clarity 

so we will now use formula for simple linear regression

y = b0 + b1*x

In [70]:
# getting intercept value, this value is given by python according to the data
slr.intercept_#b0

-224.0791521735932

In [71]:
# getting slope or coefficient value b1
slr.coef_#b1

array([3.54387057])

In [72]:
#now calculating the adiopose tissue(AT) ,for wait size = 28
# AT = b0 + b1*x
AT = -196.051 + (3.230*28)

In [73]:
AT # according to the data it is giving output as negative

-105.61099999999999

### Using root mean square error (rmse)

rmse ---> used to find the variation according to the given value eg. for -105

we are checking that how much accurate is the value of predicted Adipose tissue AT

In [74]:
# rmse

import numpy as np
print("rmse:",np.sqrt(np.mean(prd-test_y)**2))
# this is the formula for rmse

rmse: 1.4119523086254517
